javatests/arcs/android/common/resurrection/ResurrectionRequestTest.kt

```diff
@@ -0,0 +1,166 @@
+    }
+
+    @Test
+    fun createFromIntent() {
+        val bundle = PersistableBundle().apply {
+            putBoolean("foo", true)
+            putString("bar", "hello")
+        }
+        val keys = listOf(
+            RamDiskStorageKey("blah"),
+            RamDiskStorageKey("bleh")
+        )
+        val intent = Intent().apply {
+            action = ResurrectionRequest.ACTION_REQUEST_RESURRECTION
+            putExtra(ResurrectionRequest.EXTRA_REGISTRATION_PACKAGE_NAME, "com.google.test")
+            putExtra(ResurrectionRequest.EXTRA_REGISTRATION_CLASS_NAME, "MyService")
+            putExtra(
+                ResurrectionRequest.EXTRA_REGISTRATION_COMPONENT_TYPE,
+                ResurrectionRequest.ComponentType.Service.name
+            )
+            putExtra(
+                ResurrectionRequest.EXTRA_REGISTRATION_NOTIFIERS,
+                ArrayList(keys.map(StorageKey::toString))
+            )
+
+            putExtra(ResurrectionRequest.EXTRA_REGISTRATION_ACTION, "StartMePlz")
+            putExtra(ResurrectionRequest.EXTRA_REGISTRATION_EXTRAS, bundle)
+        }
+
+        val actual = ResurrectionRequest.createFromIntent(intent) ?: fail("Expected a request")
+
+        assertThat(actual.componentName).isEqualTo(ComponentName("com.google.test", "MyService"))
+        assertThat(actual.componentType).isEqualTo(ResurrectionRequest.ComponentType.Service)
+        assertThat(actual.intentAction).isEqualTo("StartMePlz")
+        assertThat(actual.intentExtras?.getBoolean("foo")).isTrue()
+        assertThat(actual.intentExtras?.getString("bar")).isEqualTo("hello")
+        assertThat(actual.notifyOn).containsExactlyElementsIn(keys)
+    }
+
+    @Test
+    fun roundtrip_fromRequest_toIntent_toRequest() {
+        val keys = listOf(
+            RamDiskStorageKey("blah"),
+            RamDiskStorageKey("bleh")
+        )
+        val request = ResurrectionRequest.createDefault(activity.activity, keys)
+        val intent = Intent().also(request::populateRequestIntent)
+        val actual = ResurrectionRequest.createFromIntent(intent)
+
+        assertThat(actual).isEqualTo(request)
+    }
+
+    @Test
+    fun createFromIntent_returnsNull_ifActionIsInvalid() {
+        val intent = Intent().apply { action = "IncorrectAction" }
+        assertThat(ResurrectionRequest.createFromIntent(intent)).isNull()
+    }
+
+    @Test
+    fun createFromIntent_returnsNull_ifNoExtras() {
+        val intent = Intent().apply {
+            action = ResurrectionRequest.ACTION_REQUEST_RESURRECTION
+            // no extras
+        }
+        assertThat(ResurrectionRequest.createFromIntent(intent)).isNull()
+    }
+
+    @Test
+    fun createFromIntent_returnsNull_ifNoPackageName() {
+        val intent = Intent().apply {
+            action = ResurrectionRequest.ACTION_REQUEST_RESURRECTION
+            // Don't include the package name in this one.
+            // putExtra(ResurrectionRequest.EXTRA_REGISTRATION_PACKAGE_NAME, "com.google.test")
```

Comment:
Totally optional, but another way of doing this would be to take a complete intent (tested in another test case), and remove the package name extra from it. Then it's both very clear what you're doing, and you don't need all the other boilerplate.

```
val intent = EXAMPLE_INTENT.copy()
    .removeExtra(ResurrectionRequest.EXTRA_REGISTRATION_PACKAGE_NAME)
```

(those are maybe not actual method names, but you get the idea)

Reply:
Ah good point, thanks! I'll use that approach in future PRs.